svgdigitizer/svgplot.py

```diff
@@ -336,10 +572,35 @@ def axis_label(axis):
             return labels[-1]
 
         return {
-            self.xlabel: axis_label(self.xlabel),
-            self.ylabel: axis_label(self.ylabel),
+            axis_symbol: axis_label(axis_symbol) for axis_symbol in self.axis_symbols
         }
 
+    @property
+    @cache
+    def _grouped_ref_points(self):
+        r"""
+        Return the reference points grouped by axis symbol.
+        """
+
+        # group ref_points by axis
+        keyfunc = lambda x: str(x.label).split(":")[0][
+            :-1
+        ]  # allow multi character axis label
+        ref_points = [points[0] for points in self.labeled_paths["ref_point"]]
+        # without sorting only the last contiguous occurences are returned in the grouping dict
```

Comment:
I think that itertools is making things more complicated here. Maybe just:

```
def variable(point): return str(x.label).split(":")[0].strip()
variables = set(variable(point) for point in ref_points)
if len(variables) != 2:
    raise ...
grouped_ref_points = { v: [point for point in ref_points if variable(point) == v] for v in variables }
```

Reply:
I agree. itertools has been removed.